documentation/docs/01-routing.md

```diff
@@ -0,0 +1,136 @@
+---
+title: Routing
+---
+
+At the heart of SvelteKit is a *filesystem-based router*. This means that the structure of your application is defined by the structure of your codebase — specifically, the contents of `src/routes`.
+
+> You can change this to a different directory by editing the [project config](#Project-configuration).
+
+There are two types of route — **pages** and **endpoints**.
+
+### Pages
+
+Pages are Svelte components written in `.svelte` files (or any file with an extension listed in [`config.extensions`](#Project-configuration)). When a user first visits the application, they will be served a server-rendered version of the page in question, plus some JavaScript that 'hydrates' the page and initialises a client-side router. From that point forward, navigating to other pages is handled entirely on the client for a fast, app-like feel.
+
+The filename determines the route. For example, `src/routes/index.svelte` is the root of your site:
+
+```html
+<!-- src/routes/index.svelte -->
+<svelte:head>
+	<title>Welcome</title>
+</svelte:head>
+
+<h1>Hello and welcome to my site!</h1>
+```
+
+A file called either `src/routes/about.svelte` or `src/routes/about/index.svelte` would correspond to the `/about` route:
+
+```html
+<!-- src/routes/about.svelte -->
+<svelte:head>
+	<title>About</title>
+</svelte:head>
+
+<h1>About this site</h1>
+<p>TODO...</p>
+```
+
+Dynamic parameters are encoded using `[brackets]`. For example, a blog post might be defined by `src/routes/blog/[slug].svelte`. Soon, we'll see how to access that parameter in a [load function](#Loading) or the [page store](#TODO).
+
+### Endpoints
+
+Endpoints are modules written in `.js` (or `.ts`) files that export functions corresponding to HTTP methods. Each function receives HTTP `request` and `context` objects as arguments. This is useful for creating a JSON API. For example our hypothetical blog page, `/blog/cool-article`, might request data from `/blog/cool-article.json`, which could be represented by a `src/routes/blog/[slug].json.js` endpoint:
+
+```js
+import db from '$lib/database';
+
+/**
+ * @param {{
```

Comment:
Is the user supposed to put this block of code on every end point? I'd expect we'd have some interface they can use

Reply:
No, it's really just there to document the types. I think it would probably be better to declare the types in a block before the code block, and then reference the types inside the code block in the way that you'd use them as an endpoint author. of course, we also need to expose those types

Reply:
added an issue for this https://github.com/sveltejs/kit/issues/547